dbhydra/dbhydra_core.py

```diff
@@ -468,12 +470,17 @@ class Mysqldb(AbstractDB):
     'datetime': "datetime"
     }
 
+    def connect_to_db(self):
+        self.connection = MySQLdb.connect(host=self.DB_SERVER, port=self.DB_PORT, user=self.DB_USERNAME, password=self.DB_PASSWORD, database=self.DB_DATABASE)
+        self.cursor = self.connection.cursor()
```

Comment:
there is `connect_to_db` implemented in `AbstractDB` that calls this method:
```    
def _connect(self):
        if self.locally:
            self.connect_locally()
        else:
            self.connect_remotely()
```
so this one seems extra

Reply:
> there is `connect_to_db` implemented in `AbstractDB` that calls this method:
> 
> ```
> def _connect(self):
>         if self.locally:
>             self.connect_locally()
>         else:
>             self.connect_remotely()
> ```
> 
> so this one seems extra

`connect_to_db` is implemented as a context manager, and called as a normal method within the `AbstractDB` init method. To my understanding, it did not make much sense (and doesn't work), so I decided to quickly hotfix it by overwriting this method in the child.

`    @contextmanager
    def connect_to_db(self):
        try:
            self._connect()
            yield None
        finally:
            self.close_connection()
`